app/cdap/components/WrangleHome/Components/OngoingDataExplorations/utils.ts

```diff
@@ -14,22 +14,34 @@
  * the License.
  */
 
-import { ImportDatasetIcon } from '../WrangleCard/iconStore/ImportDatasetIcon';
-import { IMassagedObject } from './types';
+import DataPrepStore from 'components/DataPrep/store';
+import { IEachData, IMassagedObject } from './types';
 
-export const generateDataForExplorationCard = (oldData) => {
+export const generateDataForExplorationCard = (oldData: IEachData[]) => {
   // Massaging the data to map the API response to the Ongoing Data Exploration List
   const massagedArray = [];
 
+  const { dataprep } = DataPrepStore.getState();
+  const { connectorsWithIcons } = dataprep;
+
+  const getIcon = (connectorName) => {
+    const tempItem = connectorsWithIcons.find((e) => {
```

Comment:
no need of return statement here, we can directly check the condition
const matchingConnector = connectorsWithIcons.find(eachConnector => eachConnector.name === connectorName);

Reply:
Done